File: sub-web/src/main/java/org/github/config/CorsConfig.java
package org.github.config;

import lombok.val;
import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.cors.CorsConfiguration;
import org.springframework.web.cors.UrlBasedCorsConfigurationSource;
import org.springframework.web.filter.CorsFilter;

import static org.springframework.core.Ordered.HIGHEST_PRECEDENCE;

@Configuration
public class CorsConfig {
  @Bean
  public FilterRegistrationBean<CorsFilter> corsFilter() {
    val config       = new CorsConfiguration();
    val configSource = new UrlBasedCorsConfigurationSource();
    configSource.registerCorsConfiguration("/**", config);
    val bean = new FilterRegistrationBean<>(new CorsFilter(configSource));
    bean.setOrder(HIGHEST_PRECEDENCE);
    return bean;
  }
}

File: sub-system-common/src/main/java/org/github/common/config/CorsConfig.kt
package org.github.common.config

import com.google.common.collect.ImmutableList.of
import org.springframework.boot.web.servlet.FilterRegistrationBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.Ordered.HIGHEST_PRECEDENCE
import org.springframework.web.cors.CorsConfiguration
import org.springframework.web.cors.UrlBasedCorsConfigurationSource
import org.springframework.web.filter.CorsFilter

@Configuration
class CorsConfig {
    @Bean
    fun ajaxCorsFilter(): FilterRegistrationBean<CorsFilter> {
        val config = CorsConfiguration().apply {
            allowCredentials = true
            allowedOriginPatterns = of("*")
            allowedMethods = of("*")
            allowedHeaders = of("*")
        }
        val source = UrlBasedCorsConfigurationSource().apply { registerCorsConfiguration("/**", config) }
        return FilterRegistrationBean(CorsFilter(source)).apply<FilterRegistrationBean<CorsFilter>> { order = HIGHEST_PRECEDENCE }
    }
}
